interface/patient_file/letter.php

```diff
@@ -90,6 +90,14 @@
 
 $alertmsg = ''; // anything here pops up in an alert box
 
+if (!empty($_POST)) {
+    if (!isset($_POST['token'])) {
+        error_log('WARNING: A POST request detected with no csrf token found');
+        die('Authentication failed.');
+    } else if (!hash_equals(hash_hmac('sha256', '/letter.php.theform', $_SESSION['token']), $_POST['token'])) {
+        die('Authentication failed.');
```

Comment:
Okay you die here, we will see for ajax stuffs. This should go for all forms even the ones submitted using a GET request